pkg/minikube/cluster/cluster.go

```diff
@@ -282,7 +283,19 @@ func (m *MachineConfig) CacheMinikubeISOFromURL() error {
 	}
 	defer out.Close()
 
-	if err = os.Rename(tmpISOFile.Name(), out.Name()); err != nil {
+	if err = renameFile(tmpISOFile, out); err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+func renameFile(oldFile, newFile *os.File) error {
+	// File descriptor need to be closed otherwise it will throw error
```

Comment:
I am not sure why we need another function to close the file descriptor. Whether we close it in the same function or in another function it is the same thing.